src/Microsoft.ML.CpuMath/CpuMathUtils.netcoreapp.cs

```diff
@@ -156,15 +158,16 @@ public static void MatrixTimesSource(bool transpose, AlignedArray matrix, Aligne
         /// </summary>
         /// <param name="value">The value to add.</param>
         /// <param name="destination">The destination to add the value to.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static void Add(float value, Span<float> destination)
         {
             Contracts.AssertNonEmpty(destination);
 
-            if (Avx.IsSupported)
+            if (Avx.IsSupported && destination.Length >= MinVectorSize)
```

Comment:
I would actually expect the `Avx.IsSupported` and `Sse.IsSupported` cases to have different "MinVectorSize"

Reply:
actually it tried a smaller count for sse (8) but it was still slower so i increased it to 16 and then is was faster

Reply:
Are you indicating that `Sse` for `destination.Length >= 8` was slower than `Avx` for `destination.Length >= 16`?

But for `Sse` when `destination.Length >= 16` it was faster than `Avx` for `destination.Length >= 16`?

Reply:
I am not doing a unit perf testing here. I am directly testing end to end benchmarks here. The cases I tried was
1- avx 8, sse 8
2- avx 16, sse 8
3- avx  16, sse 16

Among these, 3 gave the best results (although difference was not much difference between 2 and 3 and was within the error range)
So I am fine either way (using 2 or 3)

Reply:
I was trying to indicate that I believe that we should have two distinct constants: `SseMinVectorSize` (for the 128-bit code paths) and `AvxMinVectorSize` (for the 256-bit code paths).

I would expect that:
* Scalar is faster for 0 to X
* 128-bit is faster for X to Y
* 256-bit is faster for Y and above
Where X != Y and both X and Y are powers of 2.

Reply:
@tannergooding wont it depend upon on the operation being conducted also ? eg if we compare 2 functions one involving addition and one involving multiplication, wont the constants be different for both the cases ?